### PR TITLE
Add About eBPF guide.

### PR DESCRIPTION
**Pull request by fasaxc**
## Description

Add a general guide to eBPF and the Calico eBPF dataplane to the documentation.

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

File: about/about-ebpf.md [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+To improve performance for services, {{site.prodname}} also does connect-time load balancing by hooking into the
+socket BPF hooks.  When a program tries to connect to a Kubernetes service, {{site.prodname}} intercepts the connection
+attempt and changes and configures the socket to connect directly to the backend pod's IP instead.  This removes all
+NAT overhead from service connections.
+
+![Diagram showing BPF program attached to socket connect call; it does NAT at connect time,]({{site.baseurl}}/images/bpf-connect-time.svg "BPF program attached to socket connect call.")
+
+### Try out Calico eBPF mode
+
+If you'd like to try eBPF mode in your Kubernetes cluster, follow the [Enable the eBPF dataplane]({{site.baseurl}}/maintenance/enabling-bpf) guide.
```
Comment (lxpollitt) [inline about/about-ebpf.md:160]: Add an "above & beyond" section, with one or two links. e.g. To the blog on our eBPF performance?

File: _data/navbars/about.yml [hunk 2]
```diff
@@ -7,6 +7,8 @@ section:
   path: /about/about-calico
 - title: Networking
   path: /about/about-networking
+- title: eBPF
+  path: /about/about-ebpf
```
Comment (lxpollitt) [inline _data/navbars/about.yml:11]: Pondering whether to suggest moving this a little later in the nav order. e.g. If you don't already know about k8s networking, then some of the things discussed in this guide won't mean much.

File: about/about-ebpf.md [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+
+The logic to implement load balancing and packet parsing is pre-compiled ahead of time and relies on a set of BPF 
+maps to store the NAT frontend and backend information.  The frontend map stores the metadata of the service, allowing
+for `externalTrafficPolicy` and "sticky" services to be honoured.  NodePorts are handled as a special case: after a
+miss in the service map, a second lookup is performed to check if the destination IP is local and whether the port is an
+active node port.
+
+![Detail of BPF program showing that conntrack happens first and a conntrack hit results in taking the fast path, possibly including applying NAT to the packet.  On conntrack miss, the frontend and backend NAT maps are consulted; after a hit, the program applies NAT to the packet. The packets then take a slower path that includes the kernel's RPF check,]({{site.baseurl}}/images/bpf-nat.svg "Expanded view of tc program with conntrack and NAT.")
```
Comment (lxpollitt) [inline about/about-ebpf.md:144]: Pondering whether some of the diagrams are going into too much detail for an intro doc like this. Perhaps we could chat and explore that next week when we have a chance?

File: about/about-ebpf.md [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+- How {{site.prodname}} uses eBPF in the eBPF dataplane.
+
+### What is eBPF?
+
+eBPF is a virtual machine embedded within the Linux kernel.  It allows small programs to be loaded into the kernel,
+which are triggered when some event occurs, allowing the behaviour of the kernel to be (sometimes heavily) customised.
+While the eBPF virtual machine is the same for each type of hook, the capabilities of the hooks vary considerably.
+Since loading programs into the kernel could be dangerous; the kernel runs all programs through a very strict static 
+verifier; the verifier sandboxes the program, ensuring it can only access allowed parts of memory and ensuring that 
+it must terminate quickly.
```
[the remark this answers is not in the document]
Comment (fasaxc) [reply about/about-ebpf.md]: Think the sentence starts with something like that already

File: about/about-ebpf.md (hunk 3 above)
Comment (fasaxc) [reply about/about-ebpf.md:144]: I felt like they were getting a bit detailed but wasn't sure where to put the information if not there.

**Review by lxpollitt — APPROVED** [review]

**Pull request merged**